base/ltcounts.dtx

```diff
@@ -565,6 +559,61 @@
 %
 %
 %
+%  \begin{macro}{\counterwithout}
+%     New implementation using xparse and supporting an optional
+%    format argument.
+%    \changes{v1.1m}{2021/07/08}{New implementation for
+%          \cs{counterwithout}}
+%    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2021/11/15}%
+%<latexrelease>                 {\counterwithout}{counter without}%
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\NewDocumentCommand \counterwithout {sO{\arabic}mm}{%
+  \@ifbothcounters{#3}{#4}{%
+    \@removefromreset{#3}{#4}%
+    \IfBooleanF #1%
+        {\expandafter
+         \gdef\csname the#3\endcsname {#2{#3}}}%
```

Comment:
Does `\counterwithout` need a similar expansion on `#3` in defining `\the#3`?

```suggestion
        {\expandafter
         \xdef\csname the#3\endcsname {\unexpanded{#2}{#3}}}%
```

If so the rollbacks for `\counterwith(in|out)` need not be split.

Currently
```tex
\RequirePackage{latexbug}       % <--should be always the first line (see CONTRIBUTING)!
\documentclass{book}
\usepackage{amsmath}
\def\blub{section}

\counterwithout{\blub}{chapter}
\show\thesection
% > \thesection=macro:
% ->\arabic {\blub }.

\begin{document}
\end{document}
```

Reply:
yes it idoes need it, in fact I even made a test for it but forgot to do it then (the test shows the wrong output, sigh)

Reply:
You are right that now the rollbacks could be combined again, but I'm not going to do this it is much easier to handle individual rollbacks (if there are changes in the future) and it doesn't really hurt much if the rollback is minimally slower. We have combined ones only because initially that is much easier to set up.